operator/cmd/root.go

```diff
@@ -125,16 +125,22 @@ func Execute() {
 func registerOperatorHooks(lc hive.Lifecycle, llc *LeaderLifecycle, clientset k8sClient.Clientset, shutdowner hive.Shutdowner) {
 	apiServerShutdownSignal := make(chan struct{})
 
+	var wg sync.WaitGroup
 	lc.Append(hive.Hook{
 		OnStart: func(hive.HookContext) error {
-			go runOperator(llc, clientset, shutdowner, apiServerShutdownSignal)
+			wg.Add(1)
+			go func() {
+				runOperator(llc, clientset, shutdowner, apiServerShutdownSignal)
+				wg.Done()
+			}()
 			return nil
 		},
 		OnStop: func(ctx hive.HookContext) error {
 			if err := llc.Stop(ctx); err != nil {
 				return err
 			}
 			doCleanup()
+			wg.Wait()
```

Comment:
It is a bit weird that the waitgroup is waited before closing the channel below (this happened because that channel is no longer present in `main`). I've double-checked and this does not create issues, though, since we never wait for the apiServer termination. 

https://github.com/cilium/cilium/blob/954ffb203f39fa0754af408b1c5d098fdbcde8d8/operator/cmd/root.go#L244-L258

Reply:
yup, I was thinking about to backport a couple more PRs to make v1.13 in sync with the main branch on this part, but there is a lot :sweat: , so I just proceeded with minimum change like what we have here.